packages/fetchai/connections/p2p_libp2p/check_dependencies.py

```diff
@@ -19,19 +19,130 @@
 # ------------------------------------------------------------------------------
 
 """Check that the dependencies 'gcc' and 'go' are installed in the system."""
+import re
 import shutil
+import subprocess  # nosec
+from itertools import islice
+from typing import Any, Iterable, List, Pattern, Tuple
 
+from aea.exceptions import AEAException
 
-ERROR_MESSAGE_TEMPLATE = "{command} is required by the libp2p connection, but it is not installed, or it is not accessible from the system path."
 
+ERROR_MESSAGE_TEMPLATE_BINARY_NOT_FOUND = "'{command}' is required by the libp2p connection, but it is not installed, or it is not accessible from the system path."
+ERROR_MESSAGE_TEMPLATE_VERSION_TOO_LOW = "The installed version of '{command}' is too low: expected at least {lower_bound}; found {actual_version}."
 
-def check_binary(name: str):
-    """Check a binary is accessible from the terminal."""
-    path = shutil.which(name)
+# for the purposes of this script,
+# a version is a tuple of integers: (major, minor, patch)
+VERSION = Tuple[int, int, int]
+MINIMUM_GO_VERSION: VERSION = (1, 14, 0)
+MINIMUM_GCC_VERSION: VERSION = (7, 5, 0)
```

Comment:
Taken from:
https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu1804-README.md#language-and-runtime

However, better ask @lrahmani what is the right minimum requirement for `gcc`.